sklearn/tests/test_random_projection.py

```diff
@@ -58,7 +62,9 @@ def densify(matrix):
 
 n_samples, n_features = (10, 1000)
 n_nonzeros = int(n_samples * n_features / 100.0)
-data, data_csr = make_sparse_random_data(n_samples, n_features, n_nonzeros)
+data, data_csr = make_sparse_random_data(
+    sp.coo_array, n_samples, n_features, n_nonzeros
+)
```

Comment:
Let's remove this data generation. It means that we need to modify the following tests:

- `test_try_to_transform_before_fit`
- `test_SparseRandomProj_output_representation`
- `test_correct_RandomProjection_dimensions_embedding`
- `test_random_projection_feature_names_out`

You can call the `make_sparse_random_data` in the test itself and we will parametrize outside. For the first test (i.e., `test_try_to_transform_before_fit`), we don't have the parametrize. I think we can also avoid the parametrization on `feature_names_out`.

Reply:
Yes, now I see that the generated data is used in some tests, which I had missed before.

I have implemented it how you hinted me, @glemaitre, not parametrizing the first and the last test, but I couldn't figure out why testing the sparse martrix on them is unnecessary.